### PR TITLE
Feature/licence manager

### PR DESCRIPTION
**Pull request by davutcavdar**
Feature/licence manager

### CODE REVIEW

File: src/main/java/io/antmedia/console/AdminApplication.java [hunk 1]
```diff
@@ -224,6 +227,16 @@ public ApplicationContext getApplicationContext(String scopeName) {
 	}
 
 
+	public void updateServerSettings( ServerSettings settings) {
+		serverSettings = getServerSettings();
+
+		//	serverSettings.setServerName(settings.getServerName());
+		serverSettings.setLicenceKey(settings.getLicenceKey());
+
+		log.warn(" settings updated");
```
Comment (mekya) [inline src/main/java/io/antmedia/console/AdminApplication.java]: Please write more focused and meaningful logs like License key updated and reduce the log level. 
Log level should be info or debug

File: src/main/java/io/antmedia/console/AdminApplication.java [hunk 2]
```diff
@@ -224,6 +227,16 @@ public ApplicationContext getApplicationContext(String scopeName) {
 	}
 
 
+	public void updateServerSettings( ServerSettings settings) {
+		serverSettings = getServerSettings();
+
+		//	serverSettings.setServerName(settings.getServerName());
```
Comment (mekya) [inline src/main/java/io/antmedia/console/AdminApplication.java]: please remove the line

File: src/main/java/io/antmedia/console/rest/RestService.java [hunk 3]
```diff
@@ -64,7 +66,15 @@
 
 	public static final String IS_AUTHENTICATED = "isAuthenticated";
 
+	public static final String SERVER_NAME = "server.name";
```
Comment (mekya) [inline src/main/java/io/antmedia/console/rest/RestService.java]: afaik, server.name in properties file is "ant.media.server.name" and this key "ant.media.server.name" is used in other xml files

File: src/main/java/io/antmedia/console/rest/RestService.java [hunk 4]
```diff
@@ -697,12 +708,48 @@ public String deleteVoDStream(@PathParam("appname") String name, @FormParam("str
...
 
 
+	@POST
+	@Path("/changeServerSettings")
+	@Produces(MediaType.APPLICATION_JSON)
+	@Consumes(MediaType.APPLICATION_JSON)
+	public String changeServerSettings(ServerSettings serverSettings){
+
+
+		PreferenceStore store = new PreferenceStore("red5.properties");
```
Comment (mekya) [inline src/main/java/io/antmedia/console/rest/RestService.java]: Why we use "red5.properties" at constructor and then setFullPath?

File: src/main/java/io/antmedia/console/rest/RestService.java [hunk 5]
```diff
@@ -716,22 +763,63 @@ public Result isEnterpriseEdition(){
...
+	}
+
 	public IDataStore getDataStore() {
 		if (dataStore == null) {
 			dataStore = getDataStoreFactory().getDataStore();
 		}
 		return dataStore;
 	}
 
+	public ServerSettings fetchServerSettings() {
```
Comment (mekya) [inline src/main/java/io/antmedia/console/rest/RestService.java]: I think using a name like "getServerSettingsInternal" and make it private

File: src/main/java/io/antmedia/console/rest/RestService.java [hunk 6]
```diff
@@ -716,22 +763,63 @@ public Result isEnterpriseEdition(){
...
+	@Path("/getLicenceStatus/{key}")
+	@Produces(MediaType.APPLICATION_JSON)
+	@Consumes(MediaType.APPLICATION_JSON)
+	public Licence getLicenceStatus(@PathParam("key") String key) 
+	{
+		return getLicenceServiceInstance().checkLicence(key);
+	}
+
+
+	public void setDataStore(IDataStore dataStore) {
```
Comment (mekya) [inline src/main/java/io/antmedia/console/rest/RestService.java]: Why we need this setter? Is it used for testing?

File: src/main/java/io/antmedia/console/AdminApplication.java (hunk 2 above)
Comment (davutcavdar) [reply src/main/java/io/antmedia/console/AdminApplication.java]: removed

File: src/main/java/io/antmedia/console/AdminApplication.java (hunk 1 above)
Comment (davutcavdar) [reply src/main/java/io/antmedia/console/AdminApplication.java]: done

File: src/main/java/io/antmedia/console/rest/RestService.java (hunk 3 above)
Comment (davutcavdar) [reply src/main/java/io/antmedia/console/rest/RestService.java]: already changed

File: src/main/java/io/antmedia/console/rest/RestService.java (hunk 6 above)
Comment (davutcavdar) [reply src/main/java/io/antmedia/console/rest/RestService.java]: Yes, it is used RestServiceTest.java

File: src/main/java/io/antmedia/console/rest/RestService.java (hunk 5 above)
Comment (davutcavdar) [reply src/main/java/io/antmedia/console/rest/RestService.java]: done

**Pull request merged**